flintrock/core.py

```diff
@@ -504,6 +504,18 @@ def provision_node(
                     sudo sh -c "echo export JAVA_HOME=/usr/lib/jvm/jre >> /etc/environment"
                     source /etc/environment
                 """)
+        print("[{h}] Configuring hostname...".format(h=host))
+        ssh_check_output(
+            client=client,
+            command="""
+                set -e
+
+                fullname=`hostname`.ec2.internal
+
+                echo "{h} $fullname $(hostname)" |sudo tee -a /etc/hosts
+
+                set +e
```

Comment:
I don't think you need to do this. I believe settings like `-e` get reset between SSH sessions.